cpp/include/raft/distance/distance.cuh

```diff
@@ -252,71 +250,63 @@ void pairwise_distance(raft::resources const& handle,
                        bool isRowMajor = true,
                        Type metric_arg = 2.0f)
 {
+  cudaStream_t stream = raft::resource::get_cuda_stream(handle);
```

Comment:
It's great to see this! We should also deprecate and remove the functions that accept an optional workspace and use `raft::resource::get_workspace_resource()` (`raft/core/resource/device_memory_resource.hpp`) to instantiate a `device_uvector_policy` (currently `raft/core/detail/device_mdarray.hpp` but will become `raft/core/device_container_policy.hpp` once https://github.com/rapidsai/raft/pull/1279 is merged).

Reply:
That sounds like a good idea. Let's keep that a separate PR. That way we can keep this PR as non-breaking (as well as it's two dependent PRs).

Reply:
Created an issue to keep this one on the radar: https://github.com/rapidsai/raft/issues/1320